### PR TITLE
chore: Add rules engine to match OTEL spans

### LINKED ISSUE #2667: Build rules engine for otel synthesis
### Description
There's a process downstream that takes otel spans and synthesizes timeslice metrice, trace segments, and transaction from that data.  It has been made available as a [yml](https://source.datanerd.us/commune/apm-opentelemetry/blob/rules_engine/src/main/resources/rules.yaml). We want to copy this file for now and convert to json and start to wire up an engine that can parse this on agent connect and use it to synthesize the relevant events and metrics.

### PR DESCRIPTION
**Pull request by jsumners-nr**
This PR resolves #2667.

### CODE REVIEW

File: lib/otel/rules.test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+// Based upon https://github.com/open-telemetry/opentelemetry-js/blob/8fc76896595aac912bf9e15d4f19c167317844c8/packages/opentelemetry-sdk-trace-base/test/common/Span.test.ts#L851
+
+const test = require('node:test')
```
Comment (bizob2828) [inline lib/otel/rules.test.js:10]: is this test even getting run? it's not in the tests folder

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: lib/otel/rules.test.js (hunk 1 above)
Comment (jsumners-nr) [reply lib/otel/rules.test.js:10]: It is not getting run automatically at this time. I only wanted to be able to verify the work while we hash out the structure and underlying details. My intention is to relocate it as we draw near to it being a real feature.

Comment (bizob2828) [reply lib/otel/rules.test.js:10]: got it

**Review by bizob2828 — DISMISSED** [review]

**Comment by bizob2828** [comment]
There's a linting error

**Comment by jsumners-nr** [comment]
> There's a linting error

I know. I don't know why, though. How was I able to install and use the package if it isn't published? I'm going to shove a lint exception in there, but I have that fork in a state of change and don't want to switch branches. I'll get it updated when I get to a point where I can switch branches.

**Comment by bizob2828** [comment]
> > There's a linting error
> 
> I know. I don't know why, though. How was I able to install and use the package if it isn't published? I'm going to shove a lint exception in there, but I have that fork in a state of change and don't want to switch branches. I'll get it updated when I get to a point where I can switch branches.

Hmm, let me take a closer look, we shouldn't have to do that

**Comment by bizob2828** [comment]
Linting is failing because the test is in lib. I moved to a mirrored folder in `test/lib/otel` and the eslint rule goes away so I would just move this test file

**Comment by jsumners-nr** [comment]
> Linting is failing because the test is in lib. I moved to a mirrored folder in `test/lib/otel` and the eslint rule goes away so I would just move this test file

That's... amazing.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**